dymos/trajectory/test/test_trajectory.py

```diff
@@ -1603,6 +1603,256 @@ def compute(self, inputs, outputs):
+        start.set_state_options('v', fix_initial=False, fix_final=False)
+
+        start.add_parameter('S', units='m**2', static_target=True)
+        start.add_parameter('m', units='kg', static_target=True)
+
+        # Limit the muzzle energy
+        start.add_boundary_constraint('ke', loc='initial', upper=400000, lower=0, ref=100000)
+
+        start.add_boundary_constraint('h', loc='final', equals=2, units='m')
+
+        ascent = dm.Phase(ode_class=CannonballODE, transcription=transcription,
+                          ode_init_kwargs={'alt_unit': 'm'})
+
+        ascent = traj.add_phase('ascent', ascent)
+
+        ascent.set_time_options(fix_initial=False, duration_bounds=(1, 100),
+                                duration_ref=100, units='s')
+        ascent.set_state_options('r', fix_initial=False, fix_final=False)
+        ascent.set_state_options('h', fix_initial=False, fix_final=False)
+        ascent.set_state_options('gam', fix_initial=False, fix_final=True)
+        ascent.set_state_options('v', fix_initial=False, fix_final=False)
+
+        ascent.add_parameter('S', units='m**2', static_target=True)
+        ascent.add_parameter('m', units='kg', static_target=True)
+
+        transcription = dm.GaussLobatto(num_segments=5, order=3, compressed=True)
+        descent = dm.Phase(ode_class=CannonballODE, transcription=transcription,
+                           ode_init_kwargs={'alt_unit': 'ft'})
+
+        traj.add_phase('descent', descent)
+
+        descent.set_time_options(initial_bounds=(.5, 100), duration_bounds=(.5, 100),
+                                 duration_ref=100, units='s')
+        descent.add_state('r')
+        descent.add_state('h', fix_initial=False, fix_final=True)
+        descent.add_state('gam', fix_initial=False, fix_final=False)
+        descent.add_state('v', fix_initial=False, fix_final=False)
+
+        descent.add_parameter('S', units='m**2', static_target=True)
+        descent.add_parameter('m', units='kg', static_target=True)
+
+        descent.add_objective('r', loc='final', scaler=-1.0)
+
+        # Add internally-managed design parameters to the trajectory.
+        traj.add_parameter('CD',
+                           targets={'ascent': ['CD'], 'descent': ['CD']},
+                           val=0.5, units=None, opt=False, static_target=True)
+
+        traj.add_parameter('m', units='kg', val=1.0,
+                           targets={'ascent': 'm', 'descent': 'm'}, static_target=True)
+
+        # In this case, by omitting targets, we're connecting these
+        # parameters to parameters with the same name in each phase.
+        traj.add_parameter('S', units='m**2', val=0.005, static_target=True)
+
+        # Link Phases (link time and all state variables)
+        traj.link_phases(phases=['start', 'ascent'], vars=['*'])
+        traj.add_linkage_constraint('ascent', 'descent', 'h', 'h', 'final', 'initial', connected=True)
```

Comment:
yep, it would be more clear if I used the argument names there.  It's
`phase_a, phase_b, var_a, var_b, loc_a, loc_b`.  So here we're using `h` as the value of both sides of the equality constraint.